src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -0,0 +1,230 @@
+ // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ 
+<%@ WebHandler Language="C#" Class="PullRequestHandler" %>
+
+// This script is called to prepare a local repo for testing
+// ASSUMES: 
+//   - Git is installed on the system to "C:\Program Files\Git\bin\git.exe"
+//   - IIS web site points to this script
+//   - This script is committed to ./src/System.Private.ServiceModel/tools/PRService
+//   - 
+//   - The Git repo is set up as follows: 
+//     [remote "origin"]
+//	       url = https://github.com/dotnet/wcf
+//	       fetch = +refs/heads/*:refs/remotes/origin/*
+//	       fetch = +refs/pull/*/head:refs/remotes/origin/pr/*      <-- allows us to run git checkout pr/{id} 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Web;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Web;
+
+public class PullRequestHandler : IHttpHandler
+{
+    private const string _gitExecutablePath = @"C:\Program Files\Git\bin\git.exe";
+    private static string _gitRepoPath;
+    private const int _gitExecutionTimeoutMilliseconds = 100000;
+
+    public void ProcessRequest(HttpContext context)
+    {
+        _gitRepoPath = context.Request.PhysicalApplicationPath + "..\\..\\..\\..";
+        string prString = context.Request.QueryString["pr"];
+
+        context.Response.Write(string.Format("PR = {0}<br/>", HttpUtility.HtmlEncode(prString) ?? "Unspecified"));
+
+        StringBuilder result = new StringBuilder();
+
+        bool success = CheckPrerequisites(result);
+
+        if (!success)
+        {
+            // CheckPrerequisites failed, this is a server issue
+            context.Response.StatusCode = 500;
+            context.Response.Write(HttpUtility.HtmlEncode(result));
+            return;
+        }
+        else  
+        {
+            success = CleanupMergedBranches(result); 
+        }
+
+        if (!success)
+        {
+            // CleanupMergedBranches failed, this is a server issue
+            context.Response.StatusCode = 500;
+            context.Response.Write(HttpUtility.HtmlEncode(result));
+            return;
+        }
+        else
+        {
+            // Check inputs to the script
+            uint pr;
+            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
+            {
+                success = SyncToPr(pr, result);
+            }
+            else
+            {
+                context.Response.Write("Invalid PR ID specified<br/>");
+                success = false;
+            }
+        } 
+        
+        if (!success)
+        {
+            // it's most likely that the client supplied a bad PR ID, but there could potentially 
+            // be other errors in SyncToPr. For now, return 400
+            context.Response.StatusCode = 400;
+        }
+
+        context.Response.Write(HttpUtility.HtmlEncode(result));
```

Comment:
We don't care git output in success case. You may consider to move this line into the ```if (!success)``` clause above.